### ISSUE TITLE
Can't specify a required Header because returns it is a required query param"

### ISSUE THREAD

**Issue by DimShadoWWW** [go-fuego/fuego#238]
**To Reproduce**

Create an endpoint with a required Header:
```go
	fuego.Get(apiV1Group, "/updates", controller.GetAllUpdates,
		option.Description("Returns a list of updates since a specific timestamp."),
		option.Summary("Updates since specific timestamp"),
		option.Tags("updates"),
		option.Header("X-API-KEY", "Access token", param.Required()),
		option.Query("since", "Timestamp of the last time changes were checked", param.Required(), param.Example("Example value", "20241018T15:01:02Z")),
		option.OperationID("getUpdates"),
	)
```

Send a request:
```
curl --request GET \
  --url 'http://127.0.0.1:8080/api/v1/updates?since=20241018T15%3A01%3A02' \
  --header 'Accept: application/json' \
  --header 'X-API-KEY: testkey'
```

I get:
```
{"title":"Query Param Not Found","status":400,"detail":"cannot parse request parameter: X-API-KEY is a required query param"}
```

Instead of a correct response.

**Screenshots**

**Framework version**
fuego@v0.15.0

**Go version**
v1.23

**Comment by dylanhitt** [comment]
Ah, I forgot about Headers in the validation check. Will have a fix soon

**Comment by dylanhitt** [comment]
Implemented with #240.

**Comment by EwenQuim** [comment]
Deployed in v0.15.1. Thanks @DimShadoWWW for the report and @dylanhitt for the fix 🎉
